ports/raspberrypi/audio_dma.c

```diff
@@ -64,20 +64,23 @@ STATIC size_t audio_dma_convert_samples(audio_dma_t *dma, uint8_t *input, uint32
     if (dma->sample_resolution <= 8 && dma->output_resolution > 8) {
         // reading bytes, writing 16-bit words, so output buffer will be bigger.
 
-        output_length_used = output_length * 2;
-        if (output_length_used > output_length) {
```

Comment:
Suitable check is already occurring a few lines prior.

Also, look at what those two lines are doing… output_length_used, by definition, is now greater than output_length. The test will _always_ return true, indicating an error.

Reply:
I agree this check is wrong. However, a correct version is needed because the original check is assuming that samples are a single byte. The second check is trying (but failing) to make sure we have enough buffer for two byte samples.

I think it should be `output_length_used *= 2;`.

Reply:
That’s what’s in the PR

Reply:
Sorry, I was vague. After the times 2 you should recheck the buffer size.